src/blob-storage.js

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2022 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { GetObjectCommand, PutObjectCommand, S3 } from '@aws-sdk/client-s3';
+import { getSignedUrl } from '@aws-sdk/s3-request-presigner';
+// eslint-disable-next-line no-unused-vars
+import { Readable } from 'stream';
+
+/**
+ * @typedef BlobStorageConfig
+ * @property {Object} credentials
+ * @property {string} credentials.accessKeyId
+ * @property {string} credentials.secretAccessKey
+ * @property {string} region
+ * @property {string} bucket
+ */
+
+export class BlobStorage {
+  /**
+   * @type {string}
+   */
+  #bucket;
+
+  /**
+   * @type {S3}
+   */
+  #s3client;
+
+  /**
+   *
+   * @param {BlobStorageConfig} config
+   */
+  constructor(config) {
+    this.#s3client = config.client || new S3(config);
+    this.#bucket = config.bucket;
+  }
+
+  /**
+   * Return the binary for a blob
+   * @param {string} key - identifier of the blob
+   * @returns {Promise<Readable>}
+   */
+  async get(key) {
+    const res = await this.#s3client.send(
+      new GetObjectCommand({
+        Bucket: this.#bucket,
+        Key: key,
+      }),
+    );
+    return res.Body;
+  }
+
+  /**
+   * Return a signed URI for the blob specified by 'key' that expires in one hour.
+   * @param {string} key - identifier of the blob
+   * @returns {Promise<string>}
+   */
+  async getSignedURI(key) {
+    return getSignedUrl(
+      this.#s3client,
+      new GetObjectCommand({
+        Bucket: this.#bucket,
+        Key: key,
+      }),
+      {
+        expiresIn: 3600,
```

Comment:
nice to have: make the expiry time configurable via parameter with a default value of 3600

Reply:
I did debate just doing it, but it sounds like a feature we can add if we find a need.